app.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2016 EF CTX. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package main
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/tsuru/tsuru/cmd"
+)
+
+type createAppOptions struct {
+	name     string
+	platform string
+	team     string
+	plan     string
+	pool     string
+}
+
+func (o *createAppOptions) encode() string {
+	values := make(url.Values)
+	values.Set("name", o.name)
+	values.Set("platform", o.platform)
+	values.Set("plan", o.plan)
+	values.Set("teamOwner", o.team)
+	values.Set("pool", o.pool)
+	return values.Encode()
+}
+
+func createApp(client *cmd.Client, opts createAppOptions) (map[string]string, error) {
+	url, err := cmd.GetURL("/apps")
+	if err != nil {
+		return nil, err
+	}
+	req, err := http.NewRequest("POST", url, strings.NewReader(opts.encode()))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	var app map[string]string
+	err = json.NewDecoder(resp.Body).Decode(&app)
+	return app, err
+}
```

Comment:
The tsuru command line doesn't offer any library functions, only the commands themselves, so the commands are coupled to the shell environment. For example, this is the "function" that creates an app: https://github.com/tsuru/tsuru-client/blob/6956485efa992d03b5606ae2dd1c3359f4fde1fb/tsuru/client/apps.go#L99. So, in order to get the repository, we would either parse the output or make the app-info request.

I can revisit this and see how much we can reuse.